### PR TITLE
CORS env fix

### LINKED ISSUE #331: Review app CORS whitelist regex isn't working
I messed around with `CORS_ORIGIN_REGEX_WHITELIST` so it's different than it originally was, but I was trying to get https://foundation-mofostaging-pr-1198.herokuapp.com/fellowships/directory/ to be on the pulse staging CORS whitelist and it wasn't working. I ended up hardcoding PR-1198 into `CORS_ORIGIN_WHITELIST`.

What I'm getting at: Review apps for both pulse client and foundation site should be automatically approved for CORS.

### PR DESCRIPTION
**Pull request by patjouk**
fix #331

Why it was not working:
- there was a typo in the regex env var name,
- we don't need `r'...'` in the regex env var value on Heroku.

Since the original `django-cors-headers` lib is maintained again, I took the opportunity to switch back to it instead of using the fork.

TODO:
All those changes will require that we update the env var configuration on Heroku. Review apps and staging are sharing the same env var.

- Pulse API staging and review apps:
    - CORS_ORIGIN_WHITELIST
        - [ ] Pulse front staging
        - [ ] Pulse front dev
        - [ ] foundation staging
    - CORS_ORIGIN_REGEX_WHITELIST
        - [ ] Pulse front PR
        - [ ] foundation PR
        - [ ] localhost
        - [ ] test.example.com
    - CSRF_TRUSTED_ORIGINS
        - [ ] .herokuapp.com
- Pulse API production:
    - CORS_ORIGIN_WHITELIST
        - [ ] Pulse front prod
        - [ ] foundation prod
        - [ ] foundation staging
    - CORS_ORIGIN_REGEX_WHITELIST
        - [ ] Pulse front PR
        - [ ] foundation PR
        - [ ] localhost
        - [ ] test.example.com
    - CSRF_TRUSTED_ORIGINS
        - [ ] pulse front prod
        - [ ] foundation prod

### CODE REVIEW

**Comment by cadecairos** [comment]
@patjouk I was reading the docs on `CSRF_TRUSTED_ORIGINS`, and found this bit at the end:

>The setting also supports subdomains, so you could add ".example.com", for example, to allow access from all subdomains of example.com.

we should be able to add `".herokuapp.com"` to the list of domains in `CSRF_TRUSTED_ORIGINS` on staging to allow `POST` requests from review apps.

File: pulseapi/settings.py [hunk 1]
```diff
@@ -25,10 +25,12 @@
     DEBUG=(bool, False),
     USE_S3=(bool, False),
     SSL_PROTECTION=(bool, False),
-    CORS_REGEX_WHITELIST=(tuple, ()),
+    CORS_ORIGIN_REGEX_WHITELIST=(tuple, ()),
```
Comment (cadecairos) [inline pulseapi/settings.py]: how come this is `(tuple, ())` while the others are `(list, [])`? based on the documentation for the cors header middleware, we can set them all to be `(tuple, ())` without issue (unless I missed something)

File: pulseapi/settings.py [hunk 2]
```diff
@@ -25,10 +25,12 @@
...
+    CORS_ORIGIN_WHITELIST=(list, []),
     HEROKU_APP_NAME=(str, ''),
     PULSE_FRONTEND_HOSTNAME=(str, ''),
-    SECRET_KEY=(str, '')
+    SECRET_KEY=(str, ''),
+    CSRF_TRUSTED_ORIGINS=(list, [])
```
Comment (cadecairos) [inline pulseapi/settings.py]: small nit: lets add a trailing comma here.

File: pulseapi/settings.py [hunk 3]
```diff
@@ -231,14 +233,12 @@ def show_toolbar(request):
 CORS_ALLOW_CREDENTIALS = True
 
 # and we want origin whitelisting
-CORS_ORIGIN_WHITELIST = os.getenv(
-    'CORS_ORIGIN_WHITELIST',
-    'localhost:3000,localhost:8000,localhost:8080,test.example.com:8000,test.example.com:3000'
-).split(',')
+CORS_ORIGIN_WHITELIST = env('CORS_ORIGIN_WHITELIST')
+
+CORS_ORIGIN_REGEX_WHITELIST = env('CORS_ORIGIN_REGEX_WHITELIST')
```
Comment (cadecairos) [inline pulseapi/settings.py]: _Thank you_ for making the variable names match. this was so annoying before!

File: sample.env [hunk 4]
```diff
@@ -2,4 +2,6 @@ DEBUG=True
 REDIRECT_URIS=http://test.example.com:8000/api/pulse/oauth2callback
 SSL_PROTECTION=False
 PULSE_FRONTEND_HOSTNAME=localhost:3000
-SECRET_KEY=BetPHpGoUXUwjaAXm6ArIhV95xLdDZtu8QEGnNXY3eTknIkD
\ No newline at end of file
+SECRET_KEY=BetPHpGoUXUwjaAXm6ArIhV95xLdDZtu8QEGnNXY3eTknIkD
+CORS_ORIGIN_REGEX_WHITELIST=http://localhost:\d+,http://test.example.com:\d+
+CSRF_TRUSTED_ORIGINS=localhost:3000,localhost:8000,localhost:8080,test.example.com:8000,test.example.com:3000
```
Comment (cadecairos) [inline sample.env]: while we're touching the end of this file, add in that trailing newline.

File: pulseapi/settings.py (hunk 1 above)
Comment (patjouk) [reply pulseapi/settings.py]: It's probably me who misread the doc at some point: we can simply use list for the regex too.

**Comment by patjouk** [comment]
I tested ".herokuapp.com" yesterday and it worked. I already listed it in the var changes list :D

**Pull request merged**